src/Microsoft.AspNet.Mvc.Core/ModelBinders/BodyModelBinder.cs

```diff
@@ -57,7 +55,9 @@ public class BodyModelBinder : BindingSourceModelBinder
                 var unsupportedContentType = Resources.FormatUnsupportedContentType(
                     bindingContext.OperationBindingContext.HttpContext.Request.ContentType);
                 bindingContext.ModelState.AddModelError(bindingContext.ModelName, unsupportedContentType);
-                return new ModelBindingResult(null, bindingContext.ModelName, isModelSet: false);
+
+                // Base class will create a ModelBindingResult that exits current ModelBinding loop.
+                return null;
```

Comment:
@rynowak please stop discussing why you dislike the change and make a call on whether it's worth reverting.